pom.xml

```diff
@@ -721,6 +722,18 @@ limitations under the License.
                     </exclusion>
                 </exclusions>
             </dependency>
+            <dependency>
```

Comment:
I'm not sure what the implications are but i just checked to see if you could mock `final` classes with just mockito and came across this:

https://www.baeldung.com/mockito-final

any chance that will suffice rather than adding another testing dependency. definitely don't want to make things harder but every new bit of testing syntax seems to make it harder to get to junit5 without tons of adapters and shims (even powermock holds a `powermock-module-junit4`. Anyway, if it can't be done without the additional dependency or you see a bigger plan for testing with powermock in the future, then I suppose it's fine to keep it, otherwise my preference would be to try to do it with the test modules we already have.

Reply:
We have two requirements here. Ability to mock a final class and use spy to mock a new initialization inside method which is being tested. Using mockito to mock final classes require using an [incubating](https://github.com/mockito/mockito/wiki/What%27s-new-in-Mockito-2#unmockable) feature in mockito. I tried using that feature but it doesn't work very nicely with spy right now. Hence, powermock is the next best option.

Reply:
ok - thanks for the explanation. let's go with what you've done then. thanks

Reply:
build failures on `master` are due to powermock incompatibility with java 11 and our upgrade on the mockito version. not sure how you want to best resolve those - i tried a couple quick things but couldn't get it working immediately:

https://stackoverflow.com/questions/52966897/powermock-java-11
https://stackoverflow.com/questions/62305373/how-to-use-powermock-with-mockito3-x

I sense it is something that can be resolved though, but i'll leave it to you at this point.

Reply:
Figured out the fix i think: 8ea559e13e8e877a85c02d73178a675f31ac42b0 - builds for me locally now with `mvn clean install`.  will keep an eye on travis for the full build though.

Reply:
Thank you for helping fix this Stephen. Appreciate it 🥇